cmd/argocd-application-controller/commands/argocd_application_controller.go

```diff
@@ -212,9 +219,19 @@ func getClusterFilter(kubeClient *kubernetes.Clientset, settingsMgr *settings.Se
 			shard, err = sharding.InferShard()
 			errors.CheckError(err)
 		}
+		distributionFunction := sharding.GetShardByIdUsingHashDistributionFunction()
 		log.Infof("Processing clusters from shard %d", shard)
 		db := db.NewDB(settingsMgr.GetNamespace(), settingsMgr, kubeClient)
-		distributionFunction := sharding.GetDistributionFunction(db)
+		log.Infof("Using filter function:  %s", shardingAlgorithm)
+		switch {
+		case shardingAlgorithm == "round-robin":
+			distributionFunction = sharding.GetShardByIndexModuloReplicasCountDistributionFunction(db)
+		case shardingAlgorithm == "legacy":
+		default:
+			distributionFunctionName := runtime.FuncForPC(reflect.ValueOf(distributionFunction).Pointer())
+			log.Warnf("No distribution function named '%s' found. Defaulting to '%s'", shardingAlgorithm, distributionFunctionName)
+		}
+
```

Comment:
Why don't we re-use the `GetDistributionFunction` by sending the `shardingAlgorithm` as the parameter? If the parameter is `nil`, we continue to use the logic of the function to find the sharding algorithm, else use the value of `shardingAlgorithm`. That way, we do not have to replicate the logic at multiple places. WDYT?